iidm/iidm-serde/src/main/java/com/powsybl/iidm/serde/TerminalRefSerDe.java

```diff
@@ -75,25 +80,30 @@ private static void checkTerminal(Terminal t, NetworkSerializerContext context)
     public static Terminal readTerminal(NetworkDeserializerContext context, Network n) {
         String id = context.getAnonymizer().deanonymizeString(context.getReader().readStringAttribute(ID));
         ThreeSides side = context.getReader().readEnumAttribute(SIDE, ThreeSides.class);
+        TerminalNumber number = context.getReader().readEnumAttribute(NUMBER, TerminalNumber.class);
         context.getReader().readEndNode();
-        return TerminalRefSerDe.resolve(id, side, n);
+        return TerminalRefSerDe.resolve(id, side, number, n);
     }
 
     public static void readTerminalRef(NetworkDeserializerContext context, Network network, Consumer<Terminal> endTaskTerminalConsumer) {
         String id = context.getAnonymizer().deanonymizeString(context.getReader().readStringAttribute(ID));
         ThreeSides side = context.getReader().readEnumAttribute(SIDE, ThreeSides.class);
+        TerminalNumber number = context.getReader().readEnumAttribute(NUMBER, TerminalNumber.class);
         context.getReader().readEndNode();
         context.addEndTask(DeserializationEndTask.Step.AFTER_EXTENSIONS, () -> {
-            Terminal t = resolve(id, side, network);
+            Terminal t = resolve(id, side, number, network);
             endTaskTerminalConsumer.accept(t);
         });
     }
 
-    public static Terminal resolve(String id, ThreeSides side, Network network) {
+    public static Terminal resolve(String id, ThreeSides side, TerminalNumber number, Network network) {
         Identifiable<?> identifiable = network.getIdentifiable(id);
         if (identifiable == null) {
             throw new PowsyblException("Terminal reference identifiable not found: '" + id + "'");
         }
+        if (number != null) {
```

Comment:
When both `number` and `side` are not null, `number` has the precedence and `side` is ignored. I think it would be better to throw an exception in this case.